rest/config_test.go

```diff
@@ -440,3 +442,301 @@ func TestAutoImportEnabled(t *testing.T) {
+	os.Args = append(os.Args, "--logFilePath="+logFilePath)
+	os.Args = append(os.Args, "--pool="+pool)
+	os.Args = append(os.Args, "--pretty")
+
+	err := ParseCommandLine()
+	config := GetConfig()
+	assert.Equal(t, adminInterface, *config.AdminInterface)
+	databases := config.Databases
+	assert.Len(t, databases, 1)
+	assert.Equal(t, dbname, databases[dbname].Name)
+	assert.Equal(t, cacertpath, databases[dbname].CACertPath)
+	assert.Equal(t, certpath, databases[dbname].CertPath)
+	assert.Equal(t, keypath, databases[dbname].KeyPath)
+	assert.NoError(t, err)
+}
+
+func mockBucketConfig() BucketConfig {
+	bucket := "albums"
+	username := "Alice"
+	password := "QWxpY2U="
+
+	bucketConfig := BucketConfig{
+		Bucket:   &bucket,
+		Username: username,
+		Password: password}
+
+	return bucketConfig
+}
+
+func TestGetCredentialsFromDbConfig(t *testing.T) {
+	mockBucketConfig := mockBucketConfig()
+	dbConfig := &DbConfig{BucketConfig: mockBucketConfig}
+	username, password, bucket := dbConfig.GetCredentials()
+	assert.Equal(t, mockBucketConfig.Username, username)
+	assert.Equal(t, mockBucketConfig.Password, password)
+	assert.Equal(t, *mockBucketConfig.Bucket, bucket)
+}
+
+func TestGetCredentialsFromClusterConfig(t *testing.T) {
+	mockBucketConfig := mockBucketConfig()
+	heartbeatIntervalSeconds := uint16(10)
+	clusterConfig := &ClusterConfig{
+		BucketConfig:             mockBucketConfig,
+		DataDir:                  "/var/lib/sync_gateway/data",
+		HeartbeatIntervalSeconds: &heartbeatIntervalSeconds,
+	}
+	username, password, bucket := clusterConfig.GetCredentials()
+	assert.Equal(t, mockBucketConfig.Username, username)
+	assert.Equal(t, mockBucketConfig.Password, password)
+	assert.Equal(t, *mockBucketConfig.Bucket, bucket)
+}
+
+func TestSetMaxFileDescriptors(t *testing.T) {
+	var maxFDs uint64
+	err := SetMaxFileDescriptors(&maxFDs)
+	assert.NoError(t, err, "Error setting MaxFileDescriptors")
+
+	// Set MaxFileDescriptors to 10K;
+	maxFDs = DefaultMaxFileDescriptors * 2
+	err = SetMaxFileDescriptors(&maxFDs)
+	assert.NoError(t, err, "Error setting MaxFileDescriptors")
+}
+
+func TestParseCommandLineWithMissingConfig(t *testing.T) {
+	t.Skip("Skipping TestParseCommandLineWithMissingConfig")
+	// Parse command line options with unknown sync gateway configuration file
+	os.Args = append(os.Args, "missing-sync-gateway.conf")
+	err := ParseCommandLine()
+	config = GetConfig()
+	assert.Nil(t, config, "Configuration file doesn't exists")
+	assert.Error(t, err, "Error reading config file")
+}
+
+func createFakeConfigFile(t *testing.T, content string) string {
+	path := os.TempDir() + "sync_gateway.conf"
+	bytes := []byte(content)
+	err := ioutil.WriteFile(path, bytes, 0644)
+	assert.NoError(t, err, "Writing JSON content")
+	return path
+}
+
+func destroyFakeConfigFile(path string) {
+	var err = os.Remove(path)
+	if isError(err) {
+		return
+	}
+}
+
+func isError(err error) bool {
+	if err != nil {
+		log.Println(err.Error())
+	}
+	return err != nil
+}
```

Comment:
These wrappers seem unnessesary. Can't we call `os.Remove` from `TestParseCommandLineWithBadConfigContent` and assert on the error returned?

If it's because of the defer, we should be able to do:

```golang
defer func() {
	err := os.Remove(path)
	assert.NoError(t, err)
}()
```

Reply:
Makes sense; This change is committed to the latest version.